test/unit/iam-assume-authenticator.test.js

```diff
@@ -0,0 +1,138 @@
+  };
+
+  it('should store all config options on the class', () => {
+    const authenticator = new IamAssumeAuthenticator(config);
+
+    expect(authenticator.authenticationType()).toEqual(Authenticator.AUTHTYPE_IAM_ASSUME);
+    expect(authenticator.url).toBe(config.url);
+    expect(authenticator.clientId).toBe(config.clientId);
+    expect(authenticator.clientSecret).toBe(config.clientSecret);
+    expect(authenticator.disableSslVerification).toBe(config.disableSslVerification);
+    expect(authenticator.headers).toEqual(config.headers);
+    expect(authenticator.scope).toEqual(config.scope);
+
+    // Should also create a token manager. Note that the options like `iamProfileId`
+    // aren't stored in the authenticator, but in the token manager.
+    expect(authenticator.tokenManager).toBeInstanceOf(IamAssumeTokenManager);
+    expect(authenticator.tokenManager.iamProfileId).toBe(config.iamProfileId);
+  });
+
+  it('should throw an error when no api key is provided', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator();
+    }).toThrow('Missing required parameters: apikey');
+  });
+
+  it('should throw an error when no profile information is provided', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({ apikey: config.apikey });
+    }).toThrow(
+      'Exactly one of `iamProfileName`, `iamProfileCrn`, or `iamProfileId` must be specified.'
+    );
+  });
+
+  it('should throw an error when too much profile information is provided', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({
+        apikey: config.apikey,
+        iamProfileId: 'some-id',
+        iamProfileCrn: 'some-crn',
+      });
+    }).toThrow(
+      'Exactly one of `iamProfileName`, `iamProfileCrn`, or `iamProfileId` must be specified.'
+    );
+  });
+
+  it('should throw an error when a profile name is provided without an account id', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({
+        apikey: config.apikey,
+        iamProfileName: 'some-id',
+      });
+    }).toThrow('`iamProfileName` and `iamAccountId` must be provided together, or not at all');
+  });
+
+  it('should throw an error when an account id is provided without a profile name', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({
+        apikey: config.apikey,
+        iamProfileId: 'some-id',
+        iamAccountId: 'some-account',
+      });
+    }).toThrow('`iamProfileName` and `iamAccountId` must be provided together, or not at all');
+  });
+
+  // "End to end" style test, to make sure this authenticator ingregates properly with parent classes.
+  it('should update the options and resolve with `null` when `authenticate` is called', async () => {
+    const authenticator = new IamAssumeAuthenticator({
+      apikey: config.apikey,
+      iamProfileId: config.iamProfileId,
+    });
+
+    // Vverride the created token manager with the mocked one.
```

Comment:
```suggestion
    // Override the created token manager with the mocked one.
```
Just found another typo. 👮

Reply:
You have 🦅 👀 sir! 🙂

Reply:
Wow, how do you that?
(I had a vision correction surgery years ago and the name of the clinic where it was done is "Sasszemklinika" which means "EagleEyeClinic". 😂)

Reply:
That's actually crazy 😂 I had no idea!